File: ELEC881_Copilot/src/No Description Prompt/Java script (JS)/q25-burst-balloons.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxCoins = function(nums) {
    nums.unshift(1);
    nums.push(1);
    const n = nums.length;
    const dp = Array.from({ length: n }, () => Array(n).fill(0));
    
    for (let len = 2; len < n; len++) {
        for (let left = 0; left < n - len; left++) {
            const right = left + len;
            for (let k = left + 1; k < right; k++) {
                dp[left][right] = Math.max(
                    dp[left][right],
                    nums[left] * nums[k] * nums[right] + dp[left][k] + dp[k][right]
                );
            }
        }
    }
    
    return dp[0][n - 1];
    
};

File: ELEC881_Tabnine/src/No Description Prompt/Python/q25-burst-balloons.py
class Solution(object):
    def maxCoins(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        dp = [[0] * n for _ in range(n)]
        for length in range(2, n + 1):
            for left in range(n - length + 1):
                right = left + length - 1
                for i in range(left, right + 1):
                    dp[left][right] = max(dp[left][right], nums[left - 1] * nums[i] * nums[right + 1] + dp[left][i - 1] + dp[i + 1][right])
        return dp[0][n - 1]

